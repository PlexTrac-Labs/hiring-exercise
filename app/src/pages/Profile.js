import React, { useState, useEffect } from "react";
import axios from "axios";

import UserInfo from "../components/UserInfo";

const LOCAL_STORAGE_KEY = "app.user";
const BASE_URL = "http://localhost:5000";

export default function Profile() {
  const [user, setUser] = useState([]);

  //Load the page with the users profile
  useEffect(() => {
    const storedUser = JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEY));
    //TODO Try cath here so we don't lose the entire page
    if (storedUser && storedUser.user) handleUser(storedUser.user._id);
  }, []);

  function handleUser(key) {
    axios.get(`${BASE_URL}/user/${key}`).then(response => {
      setUser(response.data);
    });
  }

  return (
    <>
      <UserInfo user={user} />
    </>
  );
}
