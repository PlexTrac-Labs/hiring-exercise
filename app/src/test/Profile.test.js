import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Profile from "../pages/Profile";

//Test to make sure the App, entry point to everything works.
it("renders correctly", async () => {
  const tree = renderer.create(<Profile />).toJSON();
  expect(tree).toMatchSnapshot();
});
