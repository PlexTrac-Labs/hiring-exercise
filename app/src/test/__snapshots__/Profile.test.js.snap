// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="Profile"
  >
    <h2
      className="User-firstName mb-3"
    >
      John
       
      Doe
    </h2>
    <dl>
      <dt>
        User Name
      </dt>
      <dd
        className="User-userName"
      >
        jdoe
      </dd>
      <dt>
        Email
      </dt>
      <dd
        className="User-email"
      >
        johndoe@mail.com
      </dd>
      <dt>
        Year of Birth
      </dt>
      <dd
        className="User-birthYear"
      >
        1999
      </dd>
      <dt>
        Favorite Color
      </dt>
      <dd
        className="User-favColor"
      >
        Blue
      </dd>
    </dl>
  </div>
  <button
    className="btn btn-secondary"
    onClick={[Function]}
  >
    Change Info
  </button>
  <button
    className="btn btn-secondary"
    onClick={[Function]}
  >
    Change Password
  </button>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete Account
  </button>
</div>
`;
